docs/articles/documentation/reference/test-api/role/constructor.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: docs
+title: Role Constructor
+permalink: /documentation/reference/test-api/role/constructor.html
+---
+# Role Constructor
+
+Creates and initializes a [user role](../../../guides/advanced-guides/authentication.md#user-roles).
+
+```text
+Role( url, func( t ) [, options] )
+```
+
+Parameter | Type     | Description
+--------- | -------- | --------------------------------------------------------------------------------
+`url`     | String   | The URL of the login page.
+`func`    | Function | An asynchronous function that contains logic that authenticates the user.
```

Comment:
Is "contains logic" necessary here? Can we simply say:
An asynchronous function that authenticates a user.